kafka-streams/src/main/java/Account.java

```diff
@@ -0,0 +1,152 @@
+import com.google.gson.JsonObject;
+
+import java.util.Objects;
+
+public class Account {
+
+    private final String giroNumber;
+    private double balance;
+    private final String currency;
+    private long valid_from;
+    private long valid_to;
+
+    private double averageOfExpenses; //- negative change in balance
+    private int numberOfExpenses;
+
+    private Account(String giroNumber, String currency, double balance, String valid_from, String valid_to) {
+        this.giroNumber = giroNumber;
+        this.currency = currency;
+        this.balance = balance;
+        this.valid_from = CustomDateFormatter.getTimeStampFromDateInSeconds(valid_from);
+        this.valid_to = CustomDateFormatter.getTimeStampFromDateInSeconds(valid_to);
+
+        averageOfExpenses = 0;
+        numberOfExpenses = 0;
+    }
+
+    private void refreshAverageOfExpenses(double newExpenseValue) {
+        averageOfExpenses += newExpenseValue;
+        numberOfExpenses += 1;
+        averageOfExpenses = averageOfExpenses / numberOfExpenses;
+    }
+
+    public void refresh(Account newAccount) {
+        if (giroNumber.equals(newAccount.giroNumber)) {
+            balance = newAccount.balance;
+            valid_from = newAccount.valid_from;
+            valid_to = newAccount.valid_to;
+            refreshAverageOfExpenses(newAccount.balance - balance);
+        }
+    }
+
+    static Account createNewAccountFromJson(JsonObject json) {
+        String giroNumber = json.get("GIRONUMBER").getAsString();
+        double amount = Double.parseDouble(json.get("AMOUNT").getAsString());
+        String currency = json.get("CURRENCY").getAsString();
+        String valid_from_date = json.get("VALID_FROM").getAsString();
+        String valid_to_date = json.get("VALID_TO").getAsString();
+        return new Account(giroNumber,currency,amount, valid_from_date, valid_to_date);
+    }
+
+    public boolean searchPatternInExpenses(Account newAccount) {
+        boolean anomaly = false;
+        double difference = newAccount.balance - balance;
```

Comment:
From where are you taking balance here?

Reply:
It is the old balance value, coming from here: `private double balance;`